docs/api.yml

```diff
@@ -18,26 +18,61 @@ paths:
           content:
             application/json:
               schema:
-                $ref: "#/components/schemas/metadata"
+                $ref: '#/components/schemas/metadata'
               example:
                 {
-                  "documentId": "kjsak2",
-                  "firstName": "The name",
-                  "dob": "1999-01-01",
+                  'documentId': 'kjsak2',
+                  'firstName': 'The name',
+                  'dob': '1999-01-01',
                 }
   /{documentId}:
     get:
       operationId: get-metadata
       summary: Gets a document from S3 using the document id
       parameters:
-        - $ref: "#/components/parameters/documentId"
+        - $ref: '#/components/parameters/documentId'
       responses:
         200:
           description: OK
           content:
             application/json:
               schema:
-                $ref: "#/components/schemas/metadata"
+                $ref: '#/components/schemas/metadata'
+  /documents:
+    post:
+      operationId: find-documents
+      summary: Searches elasticsearch for documents with given metadata
+      requestBody:
+        content:
+          application/json:
+            schema:
+              type: object
+              example: { 'firstName': 'Toaster', 'dob': '1999-01-01' }
+      responses:
+        200:
+          description: Documents
+          content:
+            application/json:
+              example:
+                {
+                  'total': { 'value': 2, 'relation': 'eq' },
+                  'max_score': 0.5753642,
+                  'hits':
+                    [
+                      {
+                        '_index': 'documents',
+                        '_type': '_doc',
+                        '_id': 'PnJOIy',
+                        '_score': 0.5753642,
+                        '_source':
+                          {
+                            'documentId': 'PnJOIy',
+                            'firstName': 'toaster',
+                            'lastName': 'dude',
+                          },
+                      },
+                    ],
+                }
```

Comment:
Is this the raw response from ES? I wonder if it would be nicer/more useful to just have the metadata info (+ probably a link to download the file?). Just using this in doc uploads at the moment, so getting some real-world testing 😂

Reply:
Hm, I don't see much harm in returning this all info, it might be useful, would maybe make sense to structure it differently. Like { count: 2, metadata: { 'whatever is in _source' }, id: id, index: index} ? 
I think the find Docs use case would just return an array with IDs (and some extra info atm) and I'd imagine there'd be another use case: get document, which would take id and return a doc/link to a doc (?)